src/main/kotlin/uk/gov/justice/digital/hmpps/prisonertonomisupdate/personalrelationships/profiledetails/ContactPersonProfileDetailsReconciliationService.kt

```diff
@@ -63,7 +66,8 @@ class ContactPersonProfileDetailsReconciliationService(
   private suspend fun Channel<String>.waitForMismatches(): Mismatches = close().let { toList() }
 
   fun CoroutineScope.producePrisonerIds(activePrisonersCount: Long) = produce<PrisonerIds>(capacity = pageSize.toInt() * 2) {
-    (0..activePrisonersCount / pageSize)
+    val pages = ceil(activePrisonersCount * 1.0 / pageSize).toLong()
+    (0..pages - 1)
```

Comment:
There was another bug where we'd ask for an extra page if the last page was full.

Reply:
Is the * 1.0 to convert to a real number? Would using e.g. toDouble() be clearer?

Reply:
OK, next PR